main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/HarlamovBuldog/social-tournament-service/pkg/server"
+	"github.com/HarlamovBuldog/social-tournament-service/pkg/storage"
+	"gopkg.in/yaml.v3"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type config struct {
+	ConnStr string `yaml:"connstr"`
+	Port    string `yaml:"port"`
+	DBName  string `yaml:"dbname"`
+}
+
+func main() {
+	yamlConfigFile, err := ioutil.ReadFile("config.yaml")
+	if err != nil {
+		log.Printf("error opening cofiguration yaml file: %v\n", err)
+		return
+	}
+	var conf config
+	err = yaml.Unmarshal(yamlConfigFile, &conf)
+	if err != nil {
+		log.Printf("error unmarshaling yaml configuration file: %v\n", err)
+		return
+	}
+	if conf.ConnStr == "" {
+		log.Println("error in yaml configuration file: connection string is not provided")
+		return
+	}
+	clientOptions := options.Client().ApplyURI(conf.ConnStr)
+
+	client, err := mongo.Connect(context.TODO(), clientOptions)
+	defer client.Disconnect(context.TODO())
+
+	if err != nil {
+		log.Printf("error connecting to mongo db: %v", err)
+	}
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		log.Printf("error connecting to mongo db: %v", err)
+	}
+
+	log.Println("Connected to MongoDB!")
+
+	db := storage.CreateNew(client.Database("sts"))
+	s := server.NewServer(db)
+	if conf.Port == "" {
```

Comment:
so, why u checking port in the config not after parsing it? I would better create a method for the config struct, let's name it validate, inside that method check connStr and port for an empty string.